bdk_rpc_wallet_example/src/rpc.rs

```diff
@@ -0,0 +1,314 @@
+        let client = Client::new(wallet_url.as_str(), config.auth.clone().into())?;
+        let rpc_version = client.version()?;
+        println!("rpc connection established. Core version : {}", rpc_version);
+        println!("connected to '{}' with auth: {:?}", wallet_url, config.auth);
+
+        if client.list_wallets()?.contains(&wallet_name.to_string()) {
+            println!("wallet already loaded: {}", wallet_name);
+        } else if list_wallet_dirs(&client)?.contains(&wallet_name.to_string()) {
+            println!("wallet wasn't loaded. Loading wallet : {}", wallet_name);
+            client.load_wallet(&wallet_name)?;
+            println!("wallet loaded: {}", wallet_name);
+        } else {
+            // pre-0.21 use legacy wallets
+            if rpc_version < 210_000 {
+                client.create_wallet(&wallet_name, Some(true), None, None, None)?;
+            } else {
+                // TODO: move back to api call when https://github.com/rust-bitcoin/rust-bitcoincore-rpc/issues/225 is closed
+                let args = [
+                    Value::String(String::from(&wallet_name)),
+                    Value::Bool(true),  // disable_private_keys
+                    Value::Bool(false), //blank
+                    Value::Null,        // passphrase
+                    Value::Bool(false), // avoid reuse
+                    Value::Bool(false), // descriptor
+                ];
+                let _: Value = client.call("createwallet", &args)?;
+            }
+
+            println!("wallet created: {}", wallet_name);
+        }
+
+        let blockchain_info = client.get_blockchain_info()?;
+        let network = match blockchain_info.chain.as_str() {
+            "main" => Network::Bitcoin,
+            "test" => Network::Testnet,
+            "regtest" => Network::Regtest,
+            "signet" => Network::Signet,
+            _ => return Err(RpcError::General("Invalid Network string".to_string())),
+        };
+        if network != config.network {
+            return Err(RpcError::General("Invalid Network".to_string()));
+        }
+
+        import_multi(
+            &client,
+            txout_index
+                .inner()
+                .all_spks()
+                .iter()
+                .map(|(_, script)| script),
+        )?;
+
+        Ok(Self { client })
+    }
+
+    fn get_tip(&self) -> Result<(u64, BlockHash), RpcError> {
+        Ok(self
+            .client
+            .get_blockchain_info()
+            .map(|i| (i.blocks, i.best_block_hash))?)
+    }
+
+    /// Scan for a keychain tracker, and create an initial [`SparseChain`] candidate update.
+    /// This update will only contain [`Txid`]s in SparseChain, and no actual transaction data.
+    ///
+    /// User needs to fetch the required transaction data and update the [`SparseChain`] before applying it.
+    pub fn wallet_scan(
+        &self,
+        local_chain: &BTreeMap<u32, BlockHash>,
+    ) -> Result<SparseChain, RpcError> {
```

Comment:
The `SparseChain` here should at least be `SparseChain<ConfirmationTime>` since rpc returns that info I think.